src/main/java/no/unit/nva/bare/authorizer/LambdaAuthorizer.java

```diff
@@ -0,0 +1,47 @@
+package no.unit.nva.bare.authorizer;
+
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import no.unit.commons.apigateway.authentication.RequestAuthorizer;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.secrets.SecretsReader;
+
+import static nva.commons.core.attempt.Try.attempt;
+
+public class LambdaAuthorizer extends RequestAuthorizer {
+
+    public static final String DEFAULT_PRINCIPAL_ID = "ServiceAccessingBareProxy";
+    public static final String AWS_SECRET_NAME_ENV_VAR = "API_SECRET_NAME";
+    public static final String AWS_SECRET_KEY_ENV_VAR = "API_SECRET_KEY";
+    private transient final AWSSecretsManager awsSecretsManager;
+
+    @JacocoGenerated
+    public LambdaAuthorizer() {
+        this(newAwsSecretsManager(), new Environment());
+    }
+
+    public LambdaAuthorizer(AWSSecretsManager awsSecretsManager, Environment environment) {
+        super(environment);
+        this.awsSecretsManager = awsSecretsManager;
+    }
+
+    @Override
+    protected String principalId() {
+        return DEFAULT_PRINCIPAL_ID;
+    }
+
+    @Override
+    protected String fetchSecret() {
+        final String secretName = environment.readEnv(AWS_SECRET_NAME_ENV_VAR);
+        final String secretKey = environment.readEnv(AWS_SECRET_KEY_ENV_VAR);
```

Comment:
read these statically, so that we do not have to have double variables,( the name of the env variable and the value)
e.g.
```
public static final String AWS_SECRET_NAME= ENVIRONMENT.readEnv("AWS_SECRET_NAME")
```
Also, add the secrets to the "test" task for the tests and then you do not have to inject the environment